pkg/object/object.go

```diff
@@ -45,9 +48,27 @@ type OpenCompose struct {
 	Volumes  []Volume
 }
 
+func (s *Service) Validate() error {
+	// validate service name, like it cannot have underscores, etc.
+
+	// validate containers
+
+	// validate replicas
+	if s.Replicas != nil && *s.Replicas < 0 {
```

Comment:
such checks should be done in the parser as I have mentioned in my previous comments. it's n invalid input for the parser and it is the only place we can have more info like a line number - and all the other checks are done in parser as well!